Golang/README.md

```diff
@@ -0,0 +1,85 @@
+# Wikia's Golang Coding Guidelines
+
+This guidelines are new and subject to change. Our organizational experience
+with go is limited.
+
+First and foremost, read [effective
+go](https://golang.org/doc/effective_go.html). Most questions about how to
+program in go are addressed there.
+
+## Syntax and Formatting
+
+Use the [`gofmt` command](https://golang.org/doc/effective_go.html#formatting).
+It will format your code to community standard for you.
+
+## Coding Conventions
+
+### Program to Interfaces
+
+Program to an interface, not an implementation. Here is an
+example:
+
+```go
+type MediaWikiGateway interface {
+   GetArticle(articleId int) (*ArtileValue, error)
+}
+
+type ArticleServiceValue struct {
+  mediaWikiGateway MediaWikiGateway
+}
+
+func NewArticleService(gateway MediaWikiGateway) *ArticleServiceValue {
+   return &ArticleServiceValue{gateway}
+}
+
+func (articleService *ArticleServiceValue) ProcessArticle(articleId int) {
+   article, error := articleService.getArticle(articleId)
+   // do something with the article
+}
+```
+
+This allows you to mock any functionality provided by the `gateway` in the
+article service making it easier to test while also enabling dependency
+injection.
+
+### Immutability
+
+Immutable objects and programs that mutate object state are harder to reason
```

Comment:
Should this be "Mutable objects and programs that mutate object state are harder to reason about, test, and debug..." rather than how it is now which reads that "Immutable objects are harder to reason about, test, and debug..."?